src/docker.rs

```diff
@@ -978,26 +1113,33 @@ impl Docker {
     ///
     /// # API
     /// /images/json
-    pub fn images(&self, all: bool) -> Result<Vec<SummaryImage>> {
-        self.http_client()
+    pub async fn images(&self, all: bool) -> Result<Vec<SummaryImage>, DwError> {
+        let res = self
+            .http_client()
             .get(self.headers(), &format!("/images/json?a={}", all as u32))
-            .and_then(api_result)
+            .await?;
+        api_result(res).map_err(Into::into)
     }
 
     /// Get a tarball containing all images and metadata for a repository
     ///
     /// # API
     /// /images/{name}/get
-    pub fn export_image(&self, name: &str) -> Result<Box<dyn Read>> {
-        self.http_client()
-            .get(self.headers(), &format!("/images/{name}/get"))
-            .and_then(|res| {
-                if res.status.is_success() {
-                    Ok(Box::new(res) as Box<dyn Read>)
-                } else {
-                    Err(serde_json::from_reader::<_, DockerError>(res)?.into())
-                }
-            })
+    pub async fn export_image(
+        &self,
+        name: &str,
+    ) -> Result<BoxStream<'static, Result<Bytes, DwError>>, DwError> {
+        let res = self
+            .http_client()
+            .get_stream(self.headers(), &format!("/images/{name}/get"))
+            .await?;
+        if res.status().is_success() {
+            use futures::stream::StreamExt;
+            use futures::stream::TryStreamExt;
+            Ok(res.into_body().map_err(Into::into).boxed())
```

Comment:
`http::Response<hyper::Body>` -> [`impl Stream<Result<Bytes, hyper::Error>> for hyper::Body`](https://docs.rs/hyper/latest/hyper/body/struct.Body.html#impl-Stream-for-Body)